### ISSUE TITLE
Problems with module resolution in nitro

### ISSUE THREAD

**Issue by itpropro** [unjs/nitro#612]
### Environment

Node: 16.18
nitropack: latest
Windows 11 22H2 (but also tested in WSL Ubuntu)

### Reproduction

https://github.com/itpropro/nitro-repro

### Describe the bug

I am getting multiple bugs with the mongodb library, but I don't really know what they are caused by. Some, like the one in the reproduction repository are occuring in dev and production (after build), but some are only occuring after the app was build and is run with node.
The reproduction repo was the most minimal setup I could reproduce one of the errors with. They all seem so be connected to the `mongodb-connection-string-url` package (https://www.npmjs.com/package/mongodb-connection-string-url).
To reproduce the error `mongodb_connection_string_url_1.default is not a constructor`, just execute `pnpm i` and `pnpm dev` and make a POST request to `http://localhost:3000/test/123`. It should result in this:

```
mongodb_connection_string_url_1.default is not a constructor

  at parseOptions (/E:/repos/nitro-repro/.nitro/dev/index.mjs:13354:19)
  at new MongoClient (/E:/repos/nitro-repro/.nitro/dev/index.mjs:19297:64)
  at /E:/repos/nitro-repro/.nitro/dev/index.mjs:20952:29
  at /E:/repos/nitro-repro/.nitro/dev/index.mjs:635:47
  at processTicksAndRejections (node:internal/process/task_queues:96:5)
  at async Object.handler (/E:/repos/nitro-repro/.nitro/dev/index.mjs:682:19)
  at async Server.toNodeHandle (/E:/repos/nitro-repro/.nitro/dev/index.mjs:732:7)
```

with a real application, I get this error with production builds (not in dev):

```
[nitro] [request error] [unhandled] URL.install is not a function
  at Object.<anonymous> (E:\repos\project\api\.output\server\node_modules\whatwg-url\index.js:8:5)
  at Module._compile (node:internal/modules/cjs/loader:1155:14)
  at Object.Module._extensions..js (node:internal/modules/cjs/loader:1209:10)
  at Module.load (node:internal/modules/cjs/loader:1033:32)
  at Function.Module._load (node:internal/modules/cjs/loader:868:12)
  at Module.require (node:internal/modules/cjs/loader:1057:19)
  at require (node:internal/modules/cjs/helpers:103:18)
  at Object.<anonymous> (E:\repos\project\api\.output\server\node_modules\mongodb-connection-string-url\lib\index.js:4:22)
  at Module._compile (node:internal/modules/cjs/loader:1155:14)
  at Object.Module._extensions..js (node:internal/modules/cjs/loader:1209:10)
[nitro] [dev] [unhandledRejection] TypeError: URL.install is not a function
```

### Additional context

Maybe related to https://stackoverflow.com/questions/72718506/nuxt-js-3-application-with-mongoose-mongodb-not-starting-in-production

**Comment by itpropro** [comment]
Any idea or starting point I can pivot from @pi0 😄

**Comment by itpropro** [comment]
In addition, this problem seems to appear with many external libraries. E.g. the @azure/identity library works fine in dev as well, but when built, it throws and error with `[nitro] [dev] [unhandledRejection] TypeError: coreTracing.createSpanFunction is not a function` or `[nitro] [dev] [unhandledRejection] TypeError: coreTracing.createTracingClient is not a function`.
I can create another reproduction for this specific example, but I think it is a more general problem, as problems like these appear with many external libraries. Can I provide any more information that would help you with this issue @pi0 ?

**Comment by victorberland** [comment]
Facing the same issue 😔

**Comment by manniL** [comment]
Does this by any chance relate to https://github.com/unjs/jiti/issues/87?

**Comment by itpropro** [comment]
The issue is still existent with the current `nitropack-edge` version (`^2.0.0-27847631.8186a02`):

![image](https://user-images.githubusercontent.com/15030951/207628649-fa7a7cce-641e-420a-a82b-161319f34a54.png)

**Comment by itpropro** [comment]
I still have the same problems, even after xxx was merged.

None of them occur with `nitro dev`, but with `nitro build` and running it, I still get errors like:

```
coreTracing.createSpanFunction is not a function
```

With the trace being 

```
[nitro] [request error] [unhandled] coreTracing.createSpanFunction is not a function
  at Object.<anonymous> (E:\repos\project\.output\server\node_modules\@azure\app-configuration\dist\index.js:2053:32)
  at Module._compile (node:internal/modules/cjs/loader:1159:14)
  at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
  at Module.load (node:internal/modules/cjs/loader:1037:32)
  at Module._load (node:internal/modules/cjs/loader:878:12)
  at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:169:29)
  at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
  at async Promise.all (index 0)
  at async ESMLoader.import (node:internal/modules/esm/loader:530:24)
  at async Object.handler (/E:/repos/project/.output/server/chunks/node-server.mjs:455:19)
[nitro] [dev] [unhandledRejection] TypeError: coreTracing.createSpanFunction is not a function
```

Any idea @pi0? 
I will probably have to switch to Express or another HTTP server anyway, as I only got 3 months to wait for a fix from the customer and in all other tested HTTP servers, the code runs without any changes. Maybe I can come back to Nitro later, when the dependency handling works.

**Comment by itpropro** [comment]
Fixed by #909, thanks for the great work @pi0 🙏
